packages/zoe/src/contractSupport/ratio.js

```diff
@@ -61,32 +65,28 @@ export const makeRatio = (
     X`No infinite ratios! Denominator was 0/${q(denominatorBrand)}`,
   );
 
-  // TODO(https://github.com/Agoric/agoric-sdk/pull/2310) after the refactoring
-  //  use AmountMath's constructor here rather than building the record directly
   return harden({
-    numerator: { value: Nat(numerator), brand: numeratorBrand },
-    denominator: { value: Nat(denominator), brand: denominatorBrand },
+    numerator: AmountMath.make(numeratorBrand, Nat(numerator)),
+    denominator: AmountMath.make(denominatorBrand, Nat(denominator)),
   });
 };
 
+/** @type {MakeRatioFromAmounts} */
 export const makeRatioFromAmounts = (numeratorAmount, denominatorAmount) => {
-  // TODO(https://github.com/Agoric/agoric-sdk/pull/2310) after the refactoring
-  // coerce amounts using a native AmountMath operation.
-
+  AmountMath.coerce(numeratorAmount.brand, numeratorAmount);
```

Comment:
Are we coercing to check the value and the brand? If so, we might want to overwrite the `numeratorAmount` and `denominatorAmount` with their coerced values, or use an additional variable. This is what we do when we clean the offer proposal in Zoe. Basically, the uncleaned version shouldn't be relied on if we have the cleaned version available.